seed/challenges/basic-javascript.json

```diff
@@ -895,21 +895,21 @@
       "difficulty":"9.9828",
       "description":[
         "It's great that we can create random decimal numbers, but it's even more useful if we use it to generate a random whole number.",
-        "To achieve this we can multiply the random number by ten and use the <code>Math.floor()</code> to convert the decimal number to a whole number.",
+        "To achieve this we can multiply the random number by ten and use the <code>Math.floor()</code> to convert the decimal number to the nearest less than or equal whole number.",
         "This technique gives us a whole number between zero and nine.",
         "Example:",
         "<code>Math.floor(Math.random()*10);</code>",
         "Let's give this technique a go now."
       ],
       "tests":[
-        "assert(typeof(myFunction()) === \"number\", 'The result of myFunction should be a number');",
-        "assert(editor.getValue().match(/Math.random/g), 'You should be using Math.random to create a random number');",
-        "assert(!(''+myFunction()).match(/\\./g), 'You should have multiplied the result of Math.random by 10 to make it a number that\\'s greater than zero');",
-        "assert(editor.getValue().match(/Math.floor/g), 'You should use Math.floor to remove the decimal part of the number');"
+        "assert(typeof(myFunction()) === \"number\", 'The result of <code>myFunction</code> should be a number.');",
+        "assert(editor.getValue().match(/Math.random/g), 'You should be using Math.random to create a random number.');",
+        "assert(editor.getValue().match(/Math.random\\s*?\\(\\s*?\\)\\s*?\\*\\s*?10/g), 'You should have multiplied the result of <code>Math.random</code> by 10 to make it a number that\\'s between zero and nine.');",
```

Comment:
I'm not sure this regex is the best way to check if the answer is correct for two reasons:
 1. `return Math.floor(Math.random() * 100);` passes. This is the easiest to solve since you could just add the closing bracket to the regex.
 2. This is the one I'm really worried about. `return Math.floor(10 * Math.random());` doesn't pass. So you'd either have to write a really long and inconvenient regex that checks whether `10` is before or after `Math.random()` or find some other way to check the solution.